### DISCUSSION TITLE
Make Glyph.RenderTo public

### DISCUSSION THREAD

**Discussion by TareHimself** [SixLabors/Fonts#404] [Ideas] [answered]
Hi, thank you for the library. I use it to render out glyphs into an atlas. So far I have been using `TextRenderer` with a custom `IGlyphRenderer` to do this for an individual character, but now I want to support all glyphs in the font, and to do that I need access to `Glyph.RenderTo`. I would also like to request for `GlyphMetrics.ShouldSkipGlyphRendering`

**Comment by JimBobSquarePants** [comment]
I've a feeling this discussion might be useful to you. Specifically, the example given at the end which references [FontInspector](https://github.com/tocsoft/FontInspector). This demonstrates rendering each glyph within a font using the current public API.

https://github.com/SixLabors/Fonts/discussions/313

**Reply by TareHimself** [reply]
The closest method to what I am trying to achieve I noticed in FontInspector was this one https://github.com/tocsoft/FontInspector/blob/5e40e212923da4736538f9694135aa776ce7cbe3/FontInspector/Components/Glyphs.razor#L209 Which seems to still convert the codepoint into text then render it. I am working on the assumption that not all Codepoints in the font will be convertible into printable text. which is why I want to use Glyph.RenderTo. so I can do `Glyph.RenderTo(new CustomIGlyphRenderer())`

**Reply by JimBobSquarePants** [reply]
How are you getting each glyph? The methods we expose on Font require a code point. 

I don’t really want to expose that method because the arguments are the result of me hacking layout to handle various issues I found when rendering vertical text.

**Reply by TareHimself** [reply]
I do 
```csharp
var allGlyphs = font.FontMetrics.GetAvailableCodePoints().Select(c =>
        {
            font.TryGetGlyphs(c, out var glyphs);
            return glyphs?[0];
        }).Where(c => c != null).Select(c => c!.Value);
```
To get all glyphs then what I want to do is 
```csharp
using var msdfRenderer = new MtsdfRenderer();
        
var options = new TextOptions(font);
var rect = glyph.BoundingBox(GlyphLayoutMode.Horizontal, new Vector2(0.0f), options.Dpi);
msdfRenderer.BeginText(rect);
glyph.RenderTo(msdfRenderer, new Vector2(0.0f), new Vector2(0.0f), GlyphLayoutMode.Horizontal, options);

```

**Reply by TareHimself** [reply]
I just need the one glyph rendered into the IGlyphRenderer no special kerning, spacing or layout just the glyph

**Reply by TareHimself** [reply, marked as answer]
Hey, I realized that I could achieve what I wanted with the TextRenderer, sorry for the trouble, Thanks.

**Reply by JimBobSquarePants** [reply]
No worries. Glad you figured it out!
